src/NetCDFIO.jl

```diff
@@ -4,140 +4,132 @@
-    ts_grp = root_grp.group["timeseries"]
     close(root_grp)
 
     # Remove the NC file if it exists, in case it accidentally wasn't closed
     isfile(nc_filename) && rm(nc_filename; force = true)
 
     NC.Dataset(nc_filename, "c") do root_grp
         # Set profile dimensions
-        profile_grp = NC.defGroup(root_grp, "profiles")
-        NC.defDim(profile_grp, "zf", length(z_faces))
-        NC.defDim(profile_grp, "zc", length(z_centers))
-        NC.defDim(profile_grp, "t", Inf)
-        NC.defVar(profile_grp, "zf", z_faces, ("zf",))
-        NC.defVar(profile_grp, "zc", z_centers, ("zc",))
-        NC.defVar(profile_grp, "t", Float64, ("t",))
-
-        # TODO - define output variables based on the model that is being run?
-        NC.defVar(profile_grp, "density", FT, ("zc", "t"))
-        NC.defVar(profile_grp, "temperature", FT, ("zc", "t"))
-        NC.defVar(profile_grp, "pressure", FT, ("zc", "t"))
-
-        NC.defVar(profile_grp, "theta_liq_ice", FT, ("zc", "t"))
-        NC.defVar(profile_grp, "theta_dry", FT, ("zc", "t"))
-
-        NC.defVar(profile_grp, "q_tot", FT, ("zc", "t"))
-        NC.defVar(profile_grp, "q_liq", FT, ("zc", "t"))
-        NC.defVar(profile_grp, "q_ice", FT, ("zc", "t"))
-        NC.defVar(profile_grp, "q_rai", FT, ("zc", "t"))
-        NC.defVar(profile_grp, "q_sno", FT, ("zc", "t"))
+        profiles_grp = NC.defGroup(root_grp, "profiles")
+        NC.defDim(profiles_grp, "zf", length(z_faces))
+        NC.defDim(profiles_grp, "zc", length(z_centers))
+        NC.defDim(profiles_grp, "t", Inf)
+        NC.defVar(profiles_grp, "zf", z_faces, ("zf",))
+        NC.defVar(profiles_grp, "zc", z_centers, ("zc",))
+        NC.defVar(profiles_grp, "t", FT, ("t",))
+
+        for var in output_profiles
+            NC.defVar(profiles_grp, var, FT, ("zc", "t"))
+        end
 
         reference_grp = NC.defGroup(root_grp, "reference")
         NC.defDim(reference_grp, "zf", length(z_faces))
         NC.defDim(reference_grp, "zc", length(z_centers))
         NC.defVar(reference_grp, "zf", z_faces, ("zf",))
         NC.defVar(reference_grp, "zc", z_centers, ("zc",))
 
-        ts_grp = NC.defGroup(root_grp, "timeseries")
-        NC.defDim(ts_grp, "t", Inf)
-        NC.defVar(ts_grp, "t", Float64, ("t",))
+        timeseries_grp = NC.defGroup(root_grp, "timeseries")
+        NC.defDim(timeseries_grp, "t", Inf)
+        NC.defVar(timeseries_grp, "t", FT, ("t",))
+
+        #output_timeseries = ["TODO - add timeseries output"]
+        #for var in output_timeseries
+        #    NC.defVar(timeseries_grp, var, FT, ("t",))
+        #end
     end
-    vars = Dict{String, Any}()
-    return NetCDFIO_Stats(root_grp, profiles_grp, ts_grp, output_interval, nc_filename, vars)
+
+    return NetCDFIO_Stats(root_grp, output_interval, nc_filename)
 end
 
 function open_files(self::NetCDFIO_Stats)
     self.root_grp = NC.Dataset(self.nc_filename, "a")
-    self.profiles_grp = self.root_grp.group["profiles"]
```

Comment:
I would just inline (and delete `open_files` / `close_files`) this function and write it as
```julia
NC.Dataset(stats.nc_filename, "a") do ds
...
end
```